### PR TITLE
Handle Azure URLs for chinacloud

### LINKED ISSUE #7518: Support Azure Blob Storage other than default Azure Cloud
We're in the process of configuring LakeFS on Azure China Cloud. However, we've hit a snag during the setup because the management endpoint for Azure Blob Storage in China ('blob.core.chinacloudapi.cn') differs from the default Azure cloud ('blob.core.windows.net'). As a result, we're facing issues when trying to create a new repository. Is there a way to bypass this validation step in LakeFS?

图片

### PR DESCRIPTION
**Pull request by itaiad200**
closes #7518

### CODE REVIEW

**Review by N-o-Z — APPROVED** [review]
The code looks good. Two thing I think we should verify before merging:
1. Are pre-signed urls supported in this domain
2. If so, does it use the same format as in the microsoft.net domain

**Comment by itaiad200** [comment]
Good Catch @N-o-Z. We build Azure clients using hardcoded string [here](https://github.com/treeverse/lakeFS/blob/e14c7f3d6cecd4c36abf6e88b348df49ac250563/pkg/block/azure/adapter.go#L32), so we'll need to accept the url for it to work properly.

**Comment by N-o-Z** [comment]
> Good Catch @N-o-Z. We build Azure clients using hardcoded string [here](https://github.com/treeverse/lakeFS/blob/e14c7f3d6cecd4c36abf6e88b348df49ac250563/pkg/block/azure/adapter.go#L32), so we'll need to accept the url for it to work properly.

Note that it is also used in the client cache

File: pkg/block/azure/adapter.go [hunk 1]
```diff
@@ -29,14 +29,16 @@ const (
...
+	BlobEndpointGeneralFormat    = "https://%s.blob.core.windows.net/"
+	BlobEndpointChinaCloudFormat = "https://%s.blob.core.chinacloudapi.cn/"
 )
 
 type Adapter struct {
 	clientCache        *ClientCache
 	preSignedExpiry    time.Duration
 	disablePreSigned   bool
 	disablePreSignedUI bool
+	chinaCloud         bool
```
Comment (N-o-Z) [inline pkg/block/azure/adapter.go:41]: My suggestion is too not use a boolean but rather allow providing a custom domain with the default being the global Azure blob domain. In addition we might want to log a warning when using a custom domain

**Review by N-o-Z — APPROVED** [review]
Approved, but I'd like to suggest a different approach

**Comment by N-o-Z** [comment]
@itaiad200 another thing I forgot!
If I'm not mistaken there are some validations for the namespace in the GUI, we probably need to adapt those as well

**Comment by itaiad200** [comment]
> @itaiad200 another thing I forgot! If I'm not mistaken there are some validations for the namespace in the GUI, we probably need to adapt those as well

Thanks @N-o-Z . I took these into account. It's dynamically read through `ValidityRegex` & `ImportValidityRegex`. Made more changes to address it more specifically and block other domains. PTAL

**Comment by itaiad200** [comment]
Docs Preview fails due to unrelated error in one of the deps
image

**Review by N-o-Z — APPROVED** [review]

**Pull request merged**